core/templates/components/question-directives/question-player/services/question-player-state.service.ts

```diff
@@ -35,10 +35,8 @@ interface QuestionPlayerState {
   [key: string]: {
     linkedSkillIds: string[];
     answers: Answer[];
-    usedHints: UsedHint[];
-    viewedSolution: {
-      timestamp: number;
-    };
+    usedHints: UsedHintOrSolution[];
+    viewedSolution: UsedHintOrSolution | undefined;
```

Comment:
Please add a comment explaining this.

Reply:
Added comment for viewed solution being undefined